### PR TITLE
Ending Carplay on Arrival

### PR DESCRIPTION
**Pull request by JThramer**
Pic unrelated.

Like it says on the box. Possible fix for #2036.

 * Adding logic to end CarPlay navigation when we arrive at our final destination on the phone.

To-do: 
- [ ] CHANGELOG entry
- [ ] Tests?

### CODE REVIEW

File: Example/ViewController.swift [hunk 1]
```diff
@@ -412,6 +412,7 @@ extension ViewController: NavigationViewControllerDelegate {
         
         //If we're not in a "Multiple Stops" demo, show the normal EORVC
         if navigationViewController.navigationService.router.routeProgress.isFinalLeg {
+            endCarPlayNavigation(canceled: false)
```
Comment (1ec5) [inline Example/ViewController.swift]: Shouldn’t `navigationViewControllerDidDismiss(_:byCanceling:)` get called in this situation?

Comment (JThramer) [reply Example/ViewController.swift]: [It does.](https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxNavigation/CarPlayNavigationViewController.swift#L189-L199)

**Comment by 1ec5** [comment]
Fixed in #2041.

**Comment by 1ec5** [comment]
(Sorry, wrong tab. 😅)

**Pull request merged**